Taffybar crashes when window requires icon that is not found in icon theme

Errors will look like:

```
taffybar: Icon 'audio-volume-medium' not present in theme breeze (0)
```

Setting 

```
getWindowIconPixbuf = scaledWindowIconPixbufGetter getWindowIconPixbufFromEWMH
```

in WorkspacesConfig can circumvent this issue for now, but we should find a better long term fix.

@IvanMalison Having implemented your workaround and then installing every icon package under the sun (`gnome3.adwaita-icon-theme` `gnome2.gnome_icon_theme` `hicolor-icon-theme` `breeze-icons`) under both `systemPackages` and in my user environment, I'm still getting errors like

```
[WARNING] StatusNotifier.Tray - Got null pixbuf for info ItemInfo {itemServiceName = BusName ":1.161", itemServicePath = ObjectPath "/org/ayatana/NotificationItem/nm_applet", itemId = Just "nm-applet", itemStatus = Just "Active", itemCategory = Just "SystemServices", itemToolTip = Nothing, iconTitle = "Network", iconName = "nm-signal-75", overlayIconName = Nothing, iconThemePath = Just "", iconPixmaps = [], overlayIconPixmaps = [], menuPath = Just (ObjectPath "/org/ayatana/NotificationItem/nm_applet/Menu")}
```

Any ideas on what can be done to fix this with NetworkManager applet? Am I missing something?

I'd also like to mention that I get errors from `xembed`ded tray icons (using `xembed-sni-proxy`), particularly with ibus, whereby the dropdowns are buggy/buttons don't work and lots of black artifacting (some of this can be mitigated by restarting `xembed-sni-proxy`). I wonder if it's related to the many errors taffybar spews of the form

```
[ERROR] StatusNotifier.Host.Service - Property update failures [MethodError {methodErrorName = ErrorName "org.freedesktop.DBus.Error.UnknownProperty", methodErrorSerial = Serial 921, methodErrorSender = Just (BusName ":1.166"), methodErrorDestination = Just (BusName ":1.162"), methodErrorBody = [Variant "Property org.kde.StatusNotifierItem.ToolTip was not found in object /StatusNotifierItem"]}]
```

Maybe this should go in a new issue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -64,6 +64,8 @@
 
  * Center widgets will now properly expand vertically.
 
+ * Fix crash when window requires icon that is not found in icon theme.
+
 # 3.3.0
 
 ## Bug Fixes

--- a/src/System/Taffybar/Widget/Workspaces.hs
+++ b/src/System/Taffybar/Widget/Workspaces.hs
@@ -17,6 +17,7 @@ import           Control.Applicative
 import           Control.Arrow ((&&&))
 import           Control.Concurrent
 import qualified Control.Concurrent.MVar as MV
+import           Control.Exception.Enclosed (catchAny)
 import           Control.Monad
 import           Control.Monad.IO.Class
 import           Control.Monad.Trans.Class
@@ -678,16 +679,22 @@ constantScaleWindowIconPixbufGetter ::
 constantScaleWindowIconPixbufGetter constantSize getter =
   const $ scaledWindowIconPixbufGetter getter constantSize
 
+handleIconGetterException :: WindowIconPixbufGetter -> WindowIconPixbufGetter
+handleIconGetterException getter =
+  \size windowData -> catchAny (getter size windowData) $ \e -> do
+    wLog WARNING $ printf "Failed to get window icon for %s: %s" (show windowData) (show e)
+    return Nothing
+
 getWindowIconPixbufFromEWMH :: WindowIconPixbufGetter
-getWindowIconPixbufFromEWMH size windowData =
+getWindowIconPixbufFromEWMH = handleIconGetterException $ \size windowData ->
   runX11Def Nothing (getIconPixBufFromEWMH size $ windowId windowData)
 
 getWindowIconPixbufFromClass :: WindowIconPixbufGetter
-getWindowIconPixbufFromClass size windowData =
+getWindowIconPixbufFromClass = handleIconGetterException $ \size windowData ->
   lift $ getWindowIconFromClasses size (windowClass windowData)
 
 getWindowIconPixbufFromDesktopEntry :: WindowIconPixbufGetter
-getWindowIconPixbufFromDesktopEntry size windowData =
+getWindowIconPixbufFromDesktopEntry = handleIconGetterException $ \size windowData ->
   getWindowIconFromDesktopEntryByClasses size (windowClass windowData)
 
 getWindowIconPixbufFromChrome :: WindowIconPixbufGetter

